packages/p2p-v2/src/components/PaymentMethodForm/PaymentMethodForm.tsx

```diff
@@ -0,0 +1,240 @@
+                                        onClick={() => {
+                                            onAdd();
+                                            reset();
+                                        }}
+                                        width={20}
+                                    />
+                                );
+                            }}
+                            value={selectedPaymentMethod?.display_name}
+                        />
+                    ) : (
+                        <>
+                            <Dropdown
+                                label='Payment method'
+                                list={availablePaymentMethodsList}
+                                name='Payment method'
+                                onSelect={(value: string) => {
+                                    const selectedPaymentMethod = availablePaymentMethods?.find(p => p.id === value);
+                                    if (selectedPaymentMethod) {
+                                        onAdd({
+                                            displayName: selectedPaymentMethod?.display_name,
+                                            fields: selectedPaymentMethod?.fields,
+                                            method: value,
+                                        });
+                                    }
+                                }}
+                                // TODO: Remember to translate this
+                                value={selectedPaymentMethod?.display_name ?? ''}
+                                variant='comboBox'
+                            />
+                            <Text color='less-prominent' size='xs'>
+                                {/* TODO: Remember to translate these */}
+                                <span className='p2p-v2-payment-method-form__text'>Don’t see your payment method?</span>
+                                <ClickableText
+                                    color='red'
+                                    onClick={() => {
+                                        const paymentMethod = availablePaymentMethods?.find(p => p.id === 'other');
+                                        if (paymentMethod) {
+                                            onAdd({
+                                                displayName: paymentMethod?.display_name,
+                                                fields: paymentMethod?.fields,
+                                                method: 'other',
+                                            });
+                                        }
+                                    }}
+                                >
+                                    Add new.
+                                </ClickableText>
+                            </Text>
+                        </>
+                    )}
+                </div>
+                {Object.keys(selectedPaymentMethod?.fields || {})?.map(field => {
```

Comment:
but Object.keys(selectedPaymentMethod) would give error if selectedPaymentMethod is undefined.